changelog/unreleased/change-remove-click-event-oc-icon

```diff
@@ -0,0 +1,6 @@
+Change: Remove click event on OcIcon
```

Comment:
IMO not a change. Before this PR the click handler on ocIcon would be used like this:
```js
<oc-icon @click="someCallback" />
```

Since the click handler was living on the root component of the template, writing the code above would still emit a click event like before, no?

Reply:
You can still add an event handler like the `someCallback` example you provided. In theory it's a change because someone somewhere out there might have something listening on an OcIcon emitting a click event, but can re-scope the changelog item if necessary

Reply:
But the only way to listen to that `click` event is by adding a click handler like in the example above. Or could also be that I'm just missing something.